File: packages/api-types/src/models/doc-type.ts
/**
 * 文档类型(DocType)域类型。
 * DocTypeDto 来自 no.nks.dto.DocTypeDto(兼容 C# 响应形状)。
 * WrapperDocType / WrapperMultiDocTypes 来自 no.nks.entity,包裹的是 DocType 实体。
 */

/** 文档类型 DTO(C# 兼容形状)。 */
export interface DocTypeDto {
  docTypeId?: number;
  docName?: string;
  isRequired?: boolean;
}

/**
 * DocType 实体的线上形状(WrapperDocType 内包裹的类型)。
 * sortOrder / companyId 为 @JsonIgnore,不出现在线上。
 */
export interface DocType {
  id?: number;
  partyTypeId?: number;
  docName?: string;
  isRequired?: boolean;
}

/** 单个 DocType 实体包装。 */
export interface WrapperDocType {
  docType?: DocType;
}

/** 多个 DocType 实体包装。 */
export interface WrapperMultiDocTypes {
  multiDocTypes?: DocType[];
}

File: backend/src/main/java/no/nks/entity/DocType.java
package no.nks.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotEmpty;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "DocType", schema = "Party")
public class DocType {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Integer id;

    @Column(name = "PartyTypeID")
    private Integer partyTypeId;

    @NotEmpty(message = "DocName is required")
    @Column(name = "DocName")
    private String docName;

    @Column(name = "isRequired")
    private Boolean isRequired;

    @JsonIgnore
    @Column(name = "SortOrder")
    private Integer sortOrder;

    @JsonIgnore
    @Column(name = "CompanyID")
    private Integer companyId;
} 